From 316be129012555098ced9bfc117a0f33b73940b7 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 4 May 2018 18:50:38 +0800
Subject: [PATCH] ASoC: rk3308_codec: fix the missing device_unregister

We need unreigster device when we remove platform or
probe failed.

Change-Id: Ic43a9fcc540b404cee119c17c543d509b121e126
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 sound/soc/codecs/rk3308_codec.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index 68e9c45a8a3c..671f6520045c 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -2528,6 +2528,7 @@ static int rk3308_platform_probe(struct platform_device *pdev)
 	clk_disable_unprepare(rk3308->mclk_rx);
 	clk_disable_unprepare(rk3308->mclk_tx);
 	clk_disable_unprepare(rk3308->pclk);
+	device_unregister(&rk3308->dev);
 
 	return ret;
 }
@@ -2541,6 +2542,7 @@ static int rk3308_platform_remove(struct platform_device *pdev)
 	clk_disable_unprepare(rk3308->mclk_tx);
 	clk_disable_unprepare(rk3308->pclk);
 	snd_soc_unregister_codec(&pdev->dev);
+	device_unregister(&rk3308->dev);
 
 	return 0;
 }
-- 
2.35.3

